website/oauth/models/__init__.py

```diff
@@ -423,10 +423,13 @@ def refresh_oauth_key(self, force=False, extra={}, resp_auth_token_key='access_t
         except InvalidGrantError:
             raise InvalidAuthError()
         else:
-            self.account.oauth_key = token[resp_auth_token_key]
-            self.account.refresh_token = token[resp_refresh_token_key]
-            self.account.expires_at = resp_expiry_fn(token)
-            self.account.save()
+            if self.account.oauth_key != token[resp_auth_token_key]:
```

Comment:
Do you need to do this check here? It should fail and raise an exception if if the refresh request didn't work.

It shouldn't be possible if third parties follow the OAuth2 RFC, but I could imagine a poorly-constructed service that, instead of generating a new token, just extends the expiry time of the first.